src/Playwright/Transport/Connection.cs

```diff
@@ -118,30 +117,11 @@ public void Dispose()
         _callbacks.TryAdd(id, callback);
 
         var sanitizedArgs = new Dictionary<string, object>();
-        if (args != null)
+        if (dictionary != null && dictionary.Keys.Any(f => f != null))
         {
-            if (args is IDictionary<string, object> dictionary && dictionary.Keys.Any(f => f != null))
-            {
-                foreach (var kv in dictionary)
-                {
-                    if (kv.Value != null)
-                    {
-                        sanitizedArgs.Add(kv.Key, kv.Value);
-                    }
-                }
-            }
-            else
-            {
-                foreach (PropertyDescriptor propertyDescriptor in TypeDescriptor.GetProperties(args))
```

Comment:
We before were able to pass anything, this caused some issues sometimes when passing `Dictionary<string, string>` instead of `Dictionary<string, object>` which were not serialised correctly.

Reply:
Can we add a test?

Reply:
It's ensured on the type level, before we were just able to pass it now its saver.

There is still a chance that someone passes a class instance in an object, but less lower the chance. With we I mean there us.